In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

In [2]:
df=pd.read_csv("virginia semiconductors.csv")

In [3]:
df.head()

Size_of_Purchase  Company_Size  %Imports  Distance  Central_Purchaser
0              27.9          25.6        41        18                  0
1              89.6         109.8        16        75                  0
2              12.8          39.4        29        14                  0
3              34.9          16.7        31       117                  0
4             408.6         278.4        14       209                  1

In [4]:
y=df['Size_of_Purchase']
x=df.drop('Size_of_Purchase',axis=1)

In [5]:
import statsmodels.api as sm

In [6]:
Xc=sm.add_constant(x)
model=sm.OLS(y,Xc).fit()

In [7]:
model.summary()

C:\Users\ASUS\anaconda3\lib\site-packages\scipy\stats\stats.py:1603: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=17
  warnings.warn("kurtosistest only valid for n>=20 ... continuing "


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:       Size_of_Purchase   R-squared:                       0.948
Model:                            OLS   Adj. R-squared:                  0.931
Method:                 Least Squares   F-statistic:                     55.00
Date:                Mon, 25 Jan 2021   Prob (F-statistic):           1.28e-07
Time:                        12:29:45   Log-Likelihood:                -82.436
No. Observations:                  17   AIC:                             174.9
Df Residuals:                      12   BIC:                             179.0
Df Model:                           4                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                -3.6686     30.107     -0.122      0.905     -69.266      61.929
Company_Size          1.2687      0.200      6.355      0.000       0.834       1.704
%Imports              0.0508      0.865      0.059      0.954      -1.834       1.936
Distance             -0.2204      0.256     -0.860      0.407      -0.779       0.338
Central_Purchaser   127.3502     23.880      5.333      0.000      75.321     179.380
==============================================================================
Omnibus:                        2.223   Durbin-Watson:                   1.586
Prob(Omnibus):                  0.329   Jarque-Bera (JB):                1.588
Skew:                          -0.727   Prob(JB):                        0.452
Kurtosis:                       2.640   Cond. No.                         542.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

const.
p>t = 0.905
1. Ho: Constant = 0
2. Ha: Constant != 0
0.905 is the probability of getting |constant|>=-3.6686, when null hypothesis Ho is true. Hence this probability is greater than alpha (0.05), we will accept Ho.


Company_Size
const.
p>t = 0.000
1. Ho: b(Comany_Size) = 0
2. Ha: b(Company_Size) != 0
0.000 is the probability of getting |constant|>= 1.2687, when null hypothesis Ho is true. Hence this probability is less than alpha (0.05), we will reject Ho.

.
.
.

#### Lets drop the insignificant columns

In [8]:
y=df['Size_of_Purchase']
x=df.drop(['Size_of_Purchase','Distance','%Imports'],axis=1)

In [9]:
Xc=sm.add_constant(x)

In [10]:
model=sm.OLS(y,Xc).fit()

In [11]:
model.summary()

C:\Users\ASUS\anaconda3\lib\site-packages\scipy\stats\stats.py:1603: UserWarning: kurtosistest only valid for n>=20 ... continuing anyway, n=17
  warnings.warn("kurtosistest only valid for n>=20 ... continuing "


<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:       Size_of_Purchase   R-squared:                       0.942
Model:                            OLS   Adj. R-squared:                  0.934
Method:                 Least Squares   F-statistic:                     114.7
Date:                Mon, 25 Jan 2021   Prob (F-statistic):           2.09e-09
Time:                        13:41:47   Log-Likelihood:                -83.341
No. Observations:                  17   AIC:                             172.7
Df Residuals:                      14   BIC:                             175.2
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
=====================================================================================
                        coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------------
const                -8.5394     15.758     -0.542      0.596     -42.338      25.259
Company_Size          1.1961      0.146      8.182      0.000       0.883       1.510
Central_Purchaser   126.9089     21.396      5.931      0.000      81.019     172.799
==============================================================================
Omnibus:                        2.894   Durbin-Watson:                   1.423
Prob(Omnibus):                  0.235   Jarque-Bera (JB):                2.284
Skew:                          -0.830   Prob(JB):                        0.319
Kurtosis:                       2.318   Cond. No.                         320.
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""